### PR TITLE
Rename instances of the word "theme" to "mesh_library" in GridMap and MeshLibrary editors

### PR DESCRIPTION
**Pull request by YeldhamDev**

### CODE REVIEW

File: modules/gridmap/grid_map.cpp [hunk 1]
```diff
@@ -846,8 +846,8 @@ void GridMap::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("set_collision_layer_bit", "bit", "value"), &GridMap::set_collision_layer_bit);
 	ClassDB::bind_method(D_METHOD("get_collision_layer_bit", "bit"), &GridMap::get_collision_layer_bit);
 
-	ClassDB::bind_method(D_METHOD("set_theme", "theme"), &GridMap::set_theme);
-	ClassDB::bind_method(D_METHOD("get_theme"), &GridMap::get_theme);
+	ClassDB::bind_method(D_METHOD("set_mesh_library", "mesh_library"), &GridMap::set_mesh_library);
```
Comment (akien-mga) [inline modules/gridmap/grid_map.cpp]: This breaks compatibility. It's a good change but `set_theme`, `get_theme` and the `theme` property should be readded with a deprecation warning, and proxy to the `mesh_library` equivalents.

File: modules/gridmap/grid_map_editor_plugin.h [hunk 2]
```diff
@@ -187,7 +187,7 @@ class GridMapEditor : public VBoxContainer {
 	void _menu_option(int);
 	void update_pallete();
 	void _set_display_mode(int p_mode);
-	ItemList *theme_pallete;
+	ItemList *mesh_library_pallete;
```
Comment (akien-mga) [inline modules/gridmap/grid_map_editor_plugin.h]: It's actually spelled `palette`, might be worth fixing while at it :)

**Comment by YeldhamDev** [comment]
Fixed "pallete" typo and made the `*et_theme` functions warn about deprecation instead.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Great work, thanks!

**Pull request merged**